### PR TITLE
RSDK-6001 - Add Expose Paths project RPC methods

### PR DESCRIPTION
**Pull request by nicksanford**
[Ticket](https://viam.atlassian.net/browse/RSDK-6001)

Depends on: 
https://github.com/viamrobotics/rdk/pull/3303  
https://github.com/viamrobotics/viam-python-sdk/pull/496

Scope doc link: https://viam.atlassian.net/browse/RSDK-4443

- Update motion docs to describe new motion API
- Update `motion.MoveOnGlobe` description, including new request & response values & new non blocking semantics 
- Add `motion.StopPlan` description
- Add `motion.GetPlan` description
- Add `motion.ListPlanStatuses` description

### CODE REVIEW

File: docs/mobility/motion/_index.md [hunk 1]
```diff
@@ -434,6 +434,12 @@ Use a [movement sensor](/components/movement-sensor/) to check the location of t
 
 {{< alert title="Usage" color="tip" >}}
 
+`MoveOnGlobe()` is non blocking, meaning the motion service will move the component to the destination GPS point after `MoveOnGlobe()` returns.
+
+Each successful `MoveOnGlobe()` call retuns a unique ExectionID which can be used to identify all plans generated durring the `MoveOnGlobe()`.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:439]:
```suggestion
Each successful `MoveOnGlobe()` call retuns a unique ExectionID which you can use to identify all plans generated durring the `MoveOnGlobe()`.
```

File: docs/mobility/motion/_index.md [hunk 2]
```diff
@@ -434,6 +434,12 @@ Use a [movement sensor](/components/movement-sensor/) to check the location of t
...
+
+The progress of the `MoveOnGlobe()` call can be monitored by querying `GetPlan()` and `ListPlanStatuses()`.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:441]:
```suggestion
You can monitor the progress of the `MoveOnGlobe()` call by querying `GetPlan()` and `ListPlanStatuses()`.
```

File: docs/mobility/motion/_index.md [hunk 3]
```diff
@@ -537,7 +544,211 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### GetPlan
+
+Returns the plan history of the most recent [`MoveOnGlobe`](/mobility/motion/#moveonglobe) call to move a [base](/components/base/) component.
+
+Returns a result if the last execution (call to `MoveOnGlobe`) is still executing OR changed state within the last 24 hours AND the robot has not reinitialized.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:637]:
```suggestion
Returns a result if both of the following conditions are met:

- the last execution (call to `MoveOnGlobe`) is still executing **or** changed state within the last 24 hours
- the robot has not reinitialized
```

File: docs/mobility/motion/_index.md [hunk 4]
```diff
@@ -537,7 +544,211 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+
+Plans are never mutated.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:639]: That is what you're saying right? Let's say that in simpler terms.
```suggestion
Plans never change.
```

File: docs/mobility/motion/_index.md [hunk 5]
```diff
@@ -537,7 +544,211 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+    ComponentName: s.req.ComponentName,
+})
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### ListPlanStatuses
+
+Returns the plan statuses created by [`MoveOnGlobe`](/mobility/motion/#moveonglobe) calls that are executing OR are part of an execution which changed it state within the a 24HR TTL OR until the robot reinitializes.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:705]:
```suggestion
Returns the plan statuses created by [`MoveOnGlobe`](/mobility/motion/#moveonglobe) calls that meet at least one of the following conditions:

 - currently in execution
 - part of an execution which changed its state within the last 24 hours while the machine hasn't reinitialized (for example, due to a restart).
```

File: docs/mobility/motion/_index.md [hunk 6]
```diff
@@ -537,7 +544,211 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+
+All repeated fields are in time ascending order.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:707]:
```suggestion
All repeated fields are in chronological order.
```

File: docs/mobility/motion/_index.md (hunk 5 above)
Comment (nicksanford) [reply docs/mobility/motion/_index.md:708]: This would be more accurate:
```
Returns the statuses of plans created by [`MoveOnGlobe`](/mobility/motion/#moveonglobe) calls that met at least one of the following conditions since the motion service initialized:

- the plan's status is in progress
- the plan's status changed state within the last 24 hours
```

**Review by npentrel — APPROVED** [review]
Thanks!

File: docs/mobility/motion/_index.md [hunk 7]
```diff
@@ -537,7 +544,217 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+    ComponentName: s.req.ComponentName,
+})
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### GetPlan
+
+Returns the plan history of the most recent [`MoveOnGlobe`](/mobility/motion/#moveonglobe) call to move a [base](/components/base/) component.
```
Comment (nicksanford) [inline docs/mobility/motion/_index.md:635]:
```suggestion
By default: returns the plan history of the most recent [`MoveOnGlobe`](/mobility/motion/#moveonglobe) call to move a [base](/components/base/) component.

The plan history for executions before the most recent can be requested by providing an ExecutionID in the request.
```

File: docs/mobility/motion/_index.md [hunk 8]
```diff
@@ -537,7 +544,217 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+
+Returns a result if both of the following conditions are met:
+
+- the last execution (call to `MoveOnGlobe`) is still executing **or** changed state within the last 24 hours
+- the robot has not reinitialized
```
Comment (nicksanford) [inline docs/mobility/motion/_index.md:640]:
```suggestion
Returns a result if both of the following conditions are met:

- the execution (call to `MoveOnGlobe`) is still executing **or** changed state within the last 24 hours
- the robot has not reinitialized
```

**Review by nicksanford — COMMENTED** [review]
@npentrel 2 more tweaks (sorry these are somewhat nuanced APIs). LMK if those look good.

File: static/include/services/apis/motion.md [hunk 9]
```diff
@@ -5,5 +5,8 @@ Method Name | Description
 [`GetPose`](/mobility/motion/#getpose) | Get the current location and orientation of a component as a `Pose`.
 [`MoveOnMap`](/mobility/motion/#moveonmap) | Move a [base](/components/base/) component to a `Pose` in respect to the origin of a {{< glossary_tooltip term_id="slam" text="SLAM" >}} map.
 [`MoveOnGlobe`](/mobility/motion/#moveonglobe) | Move a [base](/components/base/) component to a destination GPS point. Use a [Movement Sensor](/components/movement-sensor/) to measure the robot's GPS coordinates.
+[`StopPlan`](/mobility/motion/#stopplan) | Stop a [base](/components/base/) component being moved by an in progress [`MoveOnGlobe`](/mobility/motion/#moveonglobe) call.
+[`GetPlan`](/mobility/motion/#getplan) | Returns the plan history of the most recent [`MoveOnGlobe`](/mobility/motion/#moveonglobe) call to move a [base](/components/base/) component.
+[`ListPlanStatuses`](/mobility/motion/#listplanstatuses) | Returns the plan statuses created by [`MoveOnGlobe`](/mobility/motion/#moveonglobe) calls
```
Comment (sguequierre) [inline static/include/services/apis/motion.md:10]:
```suggestion
[`ListPlanStatuses`](/mobility/motion/#listplanstatuses) | Returns the plan statuses created by [`MoveOnGlobe`](/mobility/motion/#moveonglobe) calls.
```

File: docs/mobility/motion/_index.md [hunk 10]
```diff
@@ -537,7 +544,219 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `req` [ListPlanStatusesReq](https://pkg.go.dev/go.viam.com/rdk/services/motion@v0.15.1#ListPlanStatusesReq): A `ListPlanStatusesReq` which contains the following values:
+  - `onlyActivePlans` [(bool)](https://pkg.go.dev/builtin#bool): If `true`, the response will only return plans which are executing.
+  - `extra` [(map\[string\]interface{})](https://go.dev/blog/maps): Extra options to pass to the underlying RPC call.
+
+**Returns:**
+
+- [([]PlanStatusWithID)](https://pkg.go.dev/go.viam.com/rdk/services/motion@v0.15.1#PlanStatusWithID): PlanStatusWithID describes the state of a given plan at a point in time plus the PlanId, ComponentName and ExecutionID the status is associated with.
```
Comment (sguequierre) [inline docs/mobility/motion/_index.md:750]:
```suggestion
- [([]PlanStatusWithID)](https://pkg.go.dev/go.viam.com/rdk/services/motion@v0.15.1#PlanStatusWithID): `PlanStatusWithID` describes the state of a given plan at a point in time plus the `PlanId`, `ComponentName` and `ExecutionID` the status is associated with.
```

File: docs/mobility/motion/_index.md [hunk 11]
```diff
@@ -434,6 +434,12 @@ Use a [movement sensor](/components/movement-sensor/) to check the location of t
 
 {{< alert title="Usage" color="tip" >}}
 
+`MoveOnGlobe()` is non blocking, meaning the motion service will move the component to the destination GPS point after `MoveOnGlobe()` returns.
+
+Each successful `MoveOnGlobe()` call retuns a unique ExectionID which you can use to identify all plans generated durring the `MoveOnGlobe()`.
```
Comment (sguequierre) [inline docs/mobility/motion/_index.md:439]:
```suggestion
Each successful `MoveOnGlobe()` call retuns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnGlobe()`.
```

File: docs/mobility/motion/_index.md [hunk 12]
```diff
@@ -537,7 +544,219 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+    ComponentName: s.req.ComponentName,
+})
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### GetPlan
+
+By default: returns the plan history of the most recent [`MoveOnGlobe`](/mobility/motion/#moveonglobe) call to move a [base](/components/base/) component.
```
Comment (sguequierre) [inline docs/mobility/motion/_index.md:635]:
```suggestion
By default, returns the plan history of the most recent [`MoveOnGlobe`](/mobility/motion/#moveonglobe) call to move a [base](/components/base/) component.
```

File: docs/mobility/motion/_index.md [hunk 13]
```diff
@@ -537,7 +544,219 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+
+The plan history for executions before the most recent can be requested by providing an ExecutionID in the request.
```
Comment (sguequierre) [inline docs/mobility/motion/_index.md:637]:
```suggestion
The plan history for executions before the most recent can be requested by providing an `ExecutionID` in the request.
```

File: docs/mobility/motion/_index.md [hunk 14]
```diff
@@ -537,7 +544,219 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+Returns a result if both of the following conditions are met:
+
+- the execution (call to `MoveOnGlobe`) is still executing **or** changed state within the last 24 hours
+- the robot has not reinitialized
+
+Plans never change.
+
+Replans always create new plans.
+
+Replans share the ExecutionID of the previously executing plan.
```
Comment (sguequierre) [inline docs/mobility/motion/_index.md:648]:
```suggestion
Replans share the `ExecutionID` of the previously executing plan.
```

File: docs/mobility/motion/_index.md [hunk 15]
```diff
@@ -537,7 +544,219 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+
+All repeated fields are in time ascending order.
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+**Parameters:**
+
+- `component_name` [(ResourceName)](https://python.viam.dev/autoapi/viam/gen/common/v1/common_pb2/index.html#viam.gen.common.v1.common_pb2.ResourceName): The `ResourceName` of the base to stop.
+- `lastPlanOnly` [(Optional\[bool\])](https://docs.python.org/library/typing.html#typing.Optional): If `true`, the response will only return the the last plan for the component / execution
```
Comment (sguequierre) [inline docs/mobility/motion/_index.md:658]: Could you rephrase without the slash?

File: docs/mobility/motion/_index.md [hunk 16]
```diff
@@ -537,7 +544,219 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+- `executionID` [(Optional\[str\])](https://docs.python.org/library/typing.html#typing.Optional): If non empty, the response will return the plans of the provided execution & component. Useful for retrieving plans from executions before the current execution.
+- `extra` [(Optional\[Dict\[str, Any\]\])](https://docs.python.org/library/typing.html#typing.Optional): Extra options to pass to the underlying RPC call.
+- `timeout` [(Optional\[float\])](https://docs.python.org/library/typing.html#typing.Optional): An option to set how long to wait (in seconds) before calling a time-out and closing the underlying RPC call.
+
+**Returns:**
+
+- [(GetPlanResponse)](https://python.viam.dev/autoapi/viam/services/motion/index.html#viam.services.motion.GetPlanResponse)
```
Comment (sguequierre) [inline docs/mobility/motion/_index.md:665]: Missing description

File: docs/mobility/motion/_index.md [hunk 17]
```diff
@@ -537,7 +544,219 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+
+**Parameters:**
+
+- `onlyActivePlans` [(Optional\[bool\])](https://docs.python.org/library/typing.html#typing.Optional): If `true`, the response will only return plans which are executing.
+- `extra` [(Optional\[Dict\[str, Any\]\])](https://docs.python.org/library/typing.html#typing.Optional): Extra options to pass to the underlying RPC call.
+- `timeout` [(Optional\[float\])](https://docs.python.org/library/typing.html#typing.Optional): An option to set how long to wait (in seconds) before calling a time-out and closing the underlying RPC call.
+
+**Returns:**
+
+- [(ListPlanStatusesResponse)](https://python.viam.dev/autoapi/viam/services/motion/index.html#viam.services.motion.ListPlanStatusesResponse)
```
Comment (sguequierre) [inline docs/mobility/motion/_index.md:728]: Missing description

**Review by sguequierre — CHANGES_REQUESTED** [review]
This is almost ready to merge, just a couple small things!

File: docs/mobility/motion/_index.md (hunk 15 above)
Comment (nicksanford) [reply docs/mobility/motion/_index.md]: how is that?

File: docs/mobility/motion/_index.md (hunk 16 above)
Comment (nicksanford) [reply docs/mobility/motion/_index.md]: how is that?

File: docs/mobility/motion/_index.md (hunk 17 above)
Comment (sguequierre) [reply docs/mobility/motion/_index.md:730]: I meant that there is no `: ...` text describing this return like there is for the one below

Comment (sguequierre) [reply docs/mobility/motion/_index.md:730]: Just a few words describing what the return type is

**Review by sguequierre — APPROVED** [review]
LGTM pending one last comment about return description!

File: docs/mobility/motion/_index.md [hunk 18]
```diff
@@ -537,7 +544,222 @@ myMvmntSensorResourceName := movement_sensor.Named("my_movement_sensor")
...
+  - [(plan_statuses_with_ids)](https://python.viam.dev/autoapi/viam/proto/service/motion/index.html#viam.proto.service.motion.ListPlanStatusesResponse.plan_statuses_with_ids): List of plan statuses allong with associated `planId`, `componentName` and `executionID` that match the request.
```
Comment (sguequierre) [inline docs/mobility/motion/_index.md:731]:
```suggestion
  - [(plan_statuses_with_ids)](https://python.viam.dev/autoapi/viam/proto/service/motion/index.html#viam.proto.service.motion.ListPlanStatusesResponse.plan_statuses_with_ids): List of plan statuses along with associated `planId`, `componentName` and `executionID` that match the request.
```

File: docs/mobility/motion/_index.md (hunk 17 above)
Comment (sguequierre) [reply docs/mobility/motion/_index.md:730]: I was confused about these returns being wrappers, this looks good to me!
